doc/user/content/transform-data/temporal-filters.md

```diff
@@ -0,0 +1,368 @@
+
+The temporal filter is somewhat less traditional than its non-temporal counterpart. Rather than drop or retain records ***right now***, it will schedule the insertion and deletion of records in the future.
+
+The temporal filter operator looks for predicates of the form
+
+```
+mz_now() CMP_OP EXPRESSION
+```
+
+where `CMP_OP` is a comparison operation other than `!=` (*i.e.* the operators `=`, `<`, `<=`, `>`, `>=` and things like `BETWEEN` that reduce to them) and `EXPRESSION` is an expression that does not contain `mz_now()`. Roughly, the expression is a function of `data`, and once we evaluate it we get a bound on `mz_now()`. If we have several comparisons, we end up with bounds, maybe lower and maybe upper, which describe an interval of time.
+
+An update `(data, time, diff)` normally takes effect at `time` and is in then effect indefinitely. However, we can narrow its range of time to `[lower, upper)` by transforming the input update into two output updates:
+
+```
+(data, max(time, lower), diff)
+(data, upper, -diff)
+```
+
+This change delays the insertion of `data` until at least `lower`, and schedules its deletion at `upper`.
+
+There are a variety of corner cases to double-check, mostly around what to do if a bound is absent, or if they cross (you can write it; we need to make sure it doesn't break). You'll want to double-check that the above makes sense when `diff` is negative (a deletion undoes the window its insertion would have introduced). We also need to update our query optimizer as filters can now do slightly weirder things than they could before, and it is less clear that you should use these filters ***e.g.*** to drive equi-join planning.
+
+But actually, the above is basically the implementation. The whole file comes in at around 300 lines, and that's with comments and a copyright header.
+
+## Appendix: Resource requirements
+
+Temporal filters can cause Materialize to maintain additional resources. A temporal filter evaluates a predicate on each record and can create up to two records based on the predicate, one insertion and one retraction. Many operators can process data independent of whether it's an insertion or retractions, some operators need to wait for data to be valid, i.e., their time to become current. This can lead to increased memory usage and CPU usage.
+
+This is true for arrangements, which maintain an index of data currently valid, and a stash of data that they've received but isn't valid yet. The stash is used to update the index when the data becomes valid. The size of the stash is proportional to the amount of future updates, and requires CPU time to maintain.
+
+Materialize includes optimizations to limit the amount of outstanding updates based on when they occur by dropping updates too far in the future. We still provide correctness for this optimization by ensuring we never commit any data for times that have been dropped. For wide temporal filters, this can lead to reduced CPU and memory usage.
```

Comment:
nit, but I would say "times for which updates have been dropped" rather than "times that have been dropped".